addons/point_of_sale/static/src/app/models/related_models/index.js

```diff
@@ -0,0 +1,712 @@
+                this.addItem(recordToConnect, inverse, ownerRecord);
+            }
+        }
+
+        disconnect(field, ownerRecord, recordOrIdToConnect) {
+            if (!this[STORE_SYMBOL].hasIndex(field.relation, "id")) {
+                // Not supported model
+                return;
+            }
+
+            const inverse = inverseMap.get(field);
+            const recordToDisconnect = this.getRecord(field.relation, recordOrIdToConnect);
+            if (!recordToDisconnect) {
+                return;
+            }
+
+            if (field.type === "many2one") {
+                const prevConnectedRecordId = ownerRecord[RAW_SYMBOL][field.name];
+                if (prevConnectedRecordId === recordToDisconnect.id) {
+                    ownerRecord[RAW_SYMBOL][field.name] = undefined;
+                    ownerRecord._markDirty(field.name);
+                    this.removeItem(recordToDisconnect, inverse, ownerRecord);
+                }
+            } else if (field.type === "one2many") {
+                this.removeItem(ownerRecord, field, recordToDisconnect);
+                if (!inverse.dummy) {
+                    const prevConnectedRecordId = recordToDisconnect[inverse.name]?.id;
+                    if (prevConnectedRecordId === ownerRecord.id) {
+                        recordToDisconnect[RAW_SYMBOL][inverse.name] = undefined;
+                        recordToDisconnect._markDirty(inverse.name);
+                    }
+                }
+            } else if (field.type === "many2many") {
+                this.removeItem(ownerRecord, field, recordToDisconnect);
+                this.removeItem(recordToDisconnect, inverse, ownerRecord);
+            }
+        }
+
+        getRecord(model, recordOrId) {
+            if (!recordOrId) {
+                throw new Error(`Record '${model}' is undefined`);
+            }
+            if (recordOrId instanceof Base) {
+                return recordOrId;
+            }
+            return this[STORE_SYMBOL].getById(model, recordOrId);
+        }
+
+        removeItem(ownerRecord, field, recordToRemove) {
+            if (field.dummy || !ownerRecord) {
+                return;
+            }
+            const recordIds = ownerRecord[RAW_SYMBOL][field.name];
+            const idToRemove = recordToRemove.id;
+
+            if (!idToRemove) {
+                return;
+            }
+            ownerRecord[RAW_SYMBOL][field.name] = recordIds.filter((id) => id !== idToRemove);
+            ownerRecord._markDirty(field.name);
+            ownerRecord.model.triggerEvents("update", { id: ownerRecord.id });
+        }
+
+        addItem(ownerRecord, field, recordToAdd) {
+            if (field.dummy || !ownerRecord) {
+                return;
+            }
+            const recordIds = ownerRecord[RAW_SYMBOL][field.name];
+            const idToAdd = recordToAdd.id;
+            if (!recordIds.includes(idToAdd)) {
+                recordIds.push(idToAdd);
```

Comment:
do you think we can make the `x2many` fields raw data to be `sets` in runtime? With that, we'll have the following:

```suggestion
            if (!recordIds.has(idToAdd)) {
                recordIds.add(idToAdd);
```

which results to O(1) per `addItem` call instead of O(n).